azure/services/agentpools/spec.go

```diff
@@ -160,6 +161,9 @@ func (s *AgentPoolSpec) CustomHeaders() map[string]string {
 
 // Parameters returns the parameters for the agent pool.
 func (s *AgentPoolSpec) Parameters(ctx context.Context, existing interface{}) (params interface{}, err error) {
+	_, log, done := tele.StartSpanWithLogger(ctx, "agentpools.Service.Parameters")
```

Comment:
Before when this was `ctx, log, done := ...` this was throwing an "ineffectual assignment to ctx" linter error. If this function ever needs to pass on the `ctx` it receives, then we need to remember to set `ctx` here again so we pass on the wrapped one, not the one that gets passed in as-is. Is there a good way to remind ourselves to do that? A comment here could be easily missed if `ctx` gets used deep inside this function.

Reply:
indeed https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3130